File: Stack/stack.py
class Stack:
    """Provides the base functionalities of a the stack data structure."""

    # Dunder methods

    def __init__(self) -> None:

        # The main stack list
        self._stack: list = []
        # Signifying the top of the stack (-1 means the stack is empty)
        self._size: int = -1

    def __str__(self) -> str:
        """The string format of the stack."""
        return str(f"length: {self._size}, {self._stack}")

    def __repr__(self) -> str:
        """The string representation for print and Jupyter."""
        return str(self._stack)

    # Properties

    @property
    def stack(self) -> list:
        """Returns the stack list."""
        return self._stack

    @property
    def size(self) -> int:
        """Returns the size of the stack (starts at -1 meaning the stack is emptyand goes up to the length -1)"""
        return self._size

    @property
    def top(self) -> object:
        """Returns the element at the top of the stack."""
        return self._stack[self._size]

    # Instance methods

    def push(self, value) -> None:
        """Pushes the given value argument to the top of the stack."""
        self._stack.append(value)
        self._size += 1

    def pop(self) -> object:
        """Removes and returns the top element of the stack."""
        top_element = self._stack.pop(self._size)
        self._size -= 1
        return top_element

File: Stack (C++)/Stack.cpp
#include "Stack.h"
#include <iostream>

Stack::Stack(int size)
{
	m_size = size;
	m_stack = new int[size];
	m_top = -1;
}

Stack::~Stack()
{
	delete[] m_stack;
}

void Stack::push(int value)
{
	m_top++;
	m_stack[m_top] = value;
}

int Stack::pop()
{
	int top_element = m_stack[m_top];
	m_stack[m_top] = 0;
	m_top--;
	return top_element;
}

void Stack::print()
{
	std::cout << "{ ";
	for (int i{ 0 }; i <= m_top; i++)
	{
		std::cout << m_stack[i] << " ";
	}
	std::cout << "}" << std::endl;
}

int Stack::size()
{
	return m_size;
}

int Stack::top()
{
	return m_stack[m_top];
}